ocaml/xe-cli/newcli.ml

```diff
@@ -286,6 +292,10 @@ let parse_args =
       List.rev !l
     in
     let extras_rest = process_args extras in
+    (*if traceparent is set as env var update it after we process the extras.*)
+    Option.iter
+      (fun tp -> traceparent := Some tp)
```

Comment:
The cli argument must take precedence over the environment variable, which usually takes preference over the contents of the configuration file

Reply:
This is what it does: it sets the values from configuration files, then overwrites them with the values from environment variables and then again with values from the cli argument. `process_args` calls `set_keyword` to achieve this.

This line is separate because it reads and process `TRACEPARENT` env var after it processes `XE_EXTRA_ARGS`.